server.js

```diff
@@ -175,17 +175,23 @@ Server.prototype.close = function (cb) {
   else cb(null)
 }
 
-Server.prototype.getSwarm = function (infoHash, params) {
+Server.prototype.getSwarm = function (infoHash, params, getSwarmCallback) {
   var self = this
   if (!params) params = {}
   if (Buffer.isBuffer(infoHash)) infoHash = infoHash.toString('hex')
 
-  if (self._filter && !self._filter(infoHash, params)) return null
-
-  var swarm = self.torrents[infoHash]
-  if (!swarm) swarm = self.torrents[infoHash] = new Swarm(infoHash, self)
-
-  return swarm
+  if (self._filter) {
+    self._filter(infoHash, params, function (err, res) {
+      if (err) return getSwarmCallback(err, null)
+      var swarm = self.torrents[infoHash]
+      if (!swarm) swarm = self.torrents[infoHash] = new Swarm(infoHash, self)
+      return getSwarmCallback(null, swarm)
+    })
+  } else {
+    var swarm = self.torrents[infoHash]
+    if (!swarm) swarm = self.torrents[infoHash] = new Swarm(infoHash, self)
+    return getSwarmCallback(null, swarm)
```

Comment:
Better to pull these 3 lines into an inline function declaration than to repeat them